altair/vegalite/v5/schema/_typing.py

```diff
@@ -197,6 +199,8 @@ class PaddingKwds(TypedDict, total=False):
     top: float
 
 
+Temporal: TypeAlias = Union[date, datetime]
+
```

Comment:
Imagine this new alias may spark some questions, so some context on how I landed here.

## Qs

### Why not use `date | datetime` directly?
This had two issues:
1. Adds quite a lot of additional length to the generated code.
2. The existing logic of [sort_type_reprs](https://github.com/vega/altair/blob/2ba89d508f3eac2d0bfaea53dc74a13c105a0452/tools/schemapi/utils.py#L855) produces `ann_1`, I think `ann_2` would be preferred:

```py
#                   >                     >
ann_1: str | bool | date | None | float | datetime | Parameter | SchemaBase | Map
ann_2: str | bool |  None | float | Temporal | Parameter | SchemaBase | Map
#                                   >
```
#### Side note
The doc describes something that isn't true, even prior to this PR
https://github.com/vega/altair/blob/2ba89d508f3eac2d0bfaea53dc74a13c105a0452/tools/schemapi/utils.py#L862

### Why not just use `date`?
This would pass typechecking (`datetime` derives `date`), stays lowercase and is short.

I do see it being a point of confusion though, since it requires the user knows the class hierarchy to think that `datetime` would work.

### Why the name `Temporal`?
Happy to use an alternative, but all the others I thought of weren't great:
- `DateTime` (clash with `core.DateTime`)
- `(d|D)atetimeType`
- `_datetime` (would be 2nd choice for me)
- `TemporalLiteral` 
	- (https://github.com/pola-rs/polars/blob/1d210abb0daf296aa731d3d6b955a4f18b5bb967/py-polars/polars/_typing.py#L81)